homeassistant/components/tplink/switch.py

```diff
@@ -39,7 +39,8 @@ async def async_setup_entry(
     elif device.is_plug:
         entities.append(SmartPlugSwitch(device, coordinator))
 
-    entities.append(SmartPlugLedSwitch(device, coordinator))
+    if hasattr(device, "led"):
```

Comment:
It would be nice to avoid the hasattr here because it can't be type checked

Reply:
This is a temporary solution until the we implement the interfaces (or more likely get the TAPO LED working) in python-kasa but that's not planned for this release so hopefully we can live with this until the next release?

Reply:
Please add a comment so we don't forget about it later